### PR TITLE
[MacOS]Hyperlink text working with renderconfig

### PR DESCRIPTION
**Pull request by mukuagar**
## Related Issue
Please use one of the well-known [github fixes keywords](https://help.github.com/en/articles/closing-issues-using-keywords) to reference
the issues fixed with this PR (eg Fixes #<github issue number>). If an issue doesn't yet exist please create one if reasonable to aid 
in issue tracking.

**NOTE**: For multiple issues resolved by this PR use the corresponding keywords **every time** in a comma-delimited list per the reference
page above.

## Description
Create a [descriptive title and provide all of the relevant details in the description](https://chris.beams.io/posts/git-commit/). 
This information lives with the code and will help future readers (***including yourself***) and will serve as documentation.

At the very least please describe the issue you are addressing, and what your fix entails. 

## Sample Card
image

image
Font different for first fact and second as attributed string can use cisco font meanwhile normal text can't

## How Verified
How you verified the fix, including one or all of the following: 
1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
2. Existing relevant unit/regression tests that you ran
3. Manual scenario verification if any; ***Do include .gif's or screenshots of the testing you performed here if you think that it 
will aid in code reviews or corresponding fixes on other platforms for eg.***

### CODE REVIEW

File: source/macos/AdaptiveCards/AdaptiveCards/AdaptiveCard.swift [hunk 1]
```diff
@@ -92,3 +94,19 @@ public struct ButtonConfig {
...
+    
+    let foregroundColor: NSColor
+    let isUnderlined: Bool
+    let underlineColor: NSColor
+    let underlineStyle: NSUnderlineStyle
+    
+    public init(foregroundColor: NSColor, isUnderlined: Bool, underlineColor: NSColor, underlineStyle: NSUnderlineStyle) {
+        self.foregroundColor = foregroundColor
+        self.isUnderlined = isUnderlined
+        self.underlineColor = isUnderlined ? underlineColor : .clear
```
Comment (akaashdev) [inline source/macos/AdaptiveCards/AdaptiveCards/AdaptiveCard.swift:109]: Maybe self.underlineStyle = isUnderlined ? .single : .none
Will be better than assuming cleae color here

**Review by akaashdev — COMMENTED** [review]
What about FactSetRenderer. Should we check there as well?
Also, make sure all tests pass.

File: source/macos/AdaptiveCards/AdaptiveCards/AdaptiveCard.swift (hunk 1 above)
Comment (mukuagar) [reply source/macos/AdaptiveCards/AdaptiveCards/AdaptiveCard.swift:109]: > What about FactSetRenderer. Should we check there as well?
> Also, make sure all tests pass.

Added for fact set and tests pass Akaash!

File: source/macos/AdaptiveCards/AdaptiveCards/Renderers/FactSetRenderer.swift [hunk 2]
```diff
@@ -46,8 +46,20 @@ class FactSetRenderer: NSObject, BaseCardElementRendererProtocol {
...
+                            if (attributes[.foregroundColor] as? NSColor) != nil {
+                                attributedStringCopy.addAttribute(.foregroundColor, value: textColor as Any, range: range)
+                            }
+                            if (attributes[NSAttributedString.Key.link ] as? NSURL) != nil {
+                                attributedStringCopy.addAttribute(.foregroundColor, value: config.hyperlinkColorConfig.foregroundColor, range: range)
+                                attributedStringCopy.addAttribute(.underlineColor, value: config.hyperlinkColorConfig.underlineColor, range: range)
+                                attributedStringCopy.addAttribute(.underlineStyle, value: config.hyperlinkColorConfig.underlineStyle.rawValue, range: range)
+                            }
+                        })
+                    valueView.attributedTextValue = attributedStringCopy
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Renderers/FactSetRenderer.swift:61]: Indentation issue. Line #61 is not aligned

**Pull request merged**